port/unix/omrsysinfo.c

```diff
@@ -5640,7 +5671,143 @@ readCgroupFile(struct OMRPortLibrary *portLibrary, int pid, BOOLEAN inContainer,
+ */
+static int32_t
+populateCgroupEntryListV2(struct OMRPortLibrary *portLibrary, int pid, OMRCgroupEntry **cgroupEntryList, uint64_t *availableSubsystems)
+{
+	char cgroupFilePath[PATH_MAX];
+	char controllerFilePath[PATH_MAX];
+	char buffer[PATH_MAX];
+	uintptr_t requiredSize = 0;
+	FILE *cgroupFile = NULL;
+	FILE *controllerFile = NULL;
+	OMRCgroupEntry *cgEntryList = NULL;
+	uint64_t available = 0;
+	int32_t rc = 0;
+
+	Assert_PRT_true(NULL != cgroupEntryList);
+
+	requiredSize = portLibrary->str_printf(portLibrary, NULL, 0, "/proc/%d/cgroup", pid);
+	Assert_PRT_true(requiredSize <= PATH_MAX);
+	portLibrary->str_printf(portLibrary, cgroupFilePath, sizeof(cgroupFilePath), "/proc/%d/cgroup", pid);
+
+	cgroupFile = fopen(cgroupFilePath, "r");
+	if (NULL == cgroupFile) {
+		int32_t osErrCode = errno;
+		Trc_PRT_populateCgroupEntryListV2_fopen_failed(cgroupFilePath, osErrCode);
+		rc = portLibrary->error_set_last_error(portLibrary, osErrCode, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_FOPEN_FAILED);
+		goto _end;
+	}
+
+	if (NULL == fgets(buffer, PATH_MAX, cgroupFile)) {
+		if (0 != ferror(cgroupFile)) {
+			int32_t osErrCode = errno;
+			Trc_PRT_populateCgroupEntryListV2_fgets_failed(cgroupFilePath, osErrCode);
+			rc = portLibrary->error_set_last_error_with_message_format(portLibrary, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_READ_FAILED, "fgets failed to read %s file stream with errno=%d", cgroupFilePath, osErrCode);
+			goto _end;
+		}
+	} else {
+		/* For v2, the cgroup file contains one line of the format '0::<cgroup>'. */
+		char cgroup[PATH_MAX];
+		/* This array should be large enough to read names of all subsystems. 1024 should be enough based on current supported subsystems. */
+		char subsystems[PATH_MAX];
+		char *cursor = NULL;
+		char *separator = NULL;
+
+		rc = sscanf(buffer, PROC_PID_CGROUPV2_ENTRY_FORMAT, cgroup);
+
+		if (1 == rc) {
+			/* The controller file consists of a single space-delimited line of controllers/subsystems. */
+			requiredSize = portLibrary->str_printf(portLibrary, NULL, 0, "%s/%s/cgroup.controllers", OMR_CGROUP_MOUNT_POINT, cgroup);
+			Assert_PRT_true(requiredSize <= PATH_MAX);
+			portLibrary->str_printf(portLibrary, controllerFilePath, sizeof(controllerFilePath), "%s/%s/cgroup.controllers", OMR_CGROUP_MOUNT_POINT, cgroup);
+
+			controllerFile = fopen(controllerFilePath, "r");
+			if (NULL == controllerFile) {
+				int32_t osErrCode = errno;
+				Trc_PRT_populateCgroupEntryListV2_fopen_failed(controllerFilePath, osErrCode);
+				rc = portLibrary->error_set_last_error(portLibrary, osErrCode, OMRPORT_ERROR_SYSINFO_PROCESS_CGROUP_FILE_FOPEN_FAILED);
+				goto _end;
+			}
+
+			if (NULL == fgets(subsystems, PATH_MAX, controllerFile)) {
+				/* No controllers enabled */
+				rc = 0;
+				goto _end;
+			}
+		} else {
+			Trc_PRT_populateCgroupEntryListV2_unexpected_format(cgroupFilePath);
+			rc = portLibrary->error_set_last_error_with_message_format(portLibrary,
```

Comment:
Parameter list should start on a new: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#function-parameter-list-does-not-fit-on-one-line.